x-pack/plugins/enterprise_search/public/applications/shared/licensing/license_checks.ts

```diff
@@ -0,0 +1,11 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { ILicense } from '../../../../../licensing/public';
+
+export const hasPlatinumLicense = (license: ILicense) => {
+  return license?.isActive && ['platinum', 'enterprise', 'trial'].includes(license?.type as string);
```

Comment:
Addressed in 054f659

Reply:
I should have tested this more thoroughly unfortunately - I'm getting `license is not defined` errors in the console 😬 . I'm going to have to revert this change and clarify that license is optional in the params instead of removing the optional chaining

Reply:
3b715b5